docs/madr/decisions/032-zone-to-global-policy-sync.md

```diff
@@ -0,0 +1,185 @@
+* k8s-native approach is to expect policies affect the cluster where they were applied
+* users have to use both kumactl and kubectl when Global is Universal and Zone is K8s
+* MeshGateway has to be applied on Global, while MeshGatewayInstances is created on Zone
+
+## Considered Options
+
+* Apply policies on Zone CP, scope is bounded to a single zone
+* Apply policies on Zone CP, no boundaries
+* Separate component that syncs policies from Zone to Global
+* Do nothing
+
+## Decision Outcome
+
+Apply policies on Zone CP, scope is bounded to a single zone. Optionally in the future we can implement a separate
+component that syncs policies from Zone to Global
+
+### Positive Consequences
+
+* Teams that are running Zones could be autonomous and apply policies for their services without access to Global
+* Store policies in the app repo for GitOps
+* Potential to create an automated Standalone to Zone conversion when all the policies stays on the cluster as is.
+  Application developers don't need to be aware that there was any standalone to multizone migration. From their perspective
+  the workflow stays the same. Standalone mode can be abolished, CP is Zone by default even if there is no connection to Global.
+* Policies take effect in the cluster where they were applied
+
+### Negative Consequences
+
+* Still have to use `kumactl` with access to Global when user wants to change client's outbounds in another Zone
+
+## Pros and Cons of the Options
+
+### Apply policies on Zone CP, scope is bounded to a single zone
+
+Users apply policies on Zone CP and Kuma automatically adds `kuma.io/zone` tag to the top-level targetRefs before syncing
+these policies to Global. For example, when user creates a policy in `us-east` zone:
+
+```yaml
+type: MeshTimeout
+mesh: mesh-1
+name: t-1
+labels: # not implemented today, requires Postgres schema 
+  kuma.io/managed-by: zone
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  to: [...]
+```
+
+Kuma automatically converts this to:
+
+```yaml
+type: MeshTimeout
+mesh: mesh-1
+name: t-1
+labels: # not implemented today, requires Postgres schema 
+  kuma.io/managed-by: zone
+spec:
+  targetRef:
+    kind: MeshServiceSubset
+    name: backend
+    tags:
+      kuma.io/zone: us-east
+  to: [...]
+```
+
+This conversion is visible inside the Zone store (so it happens **not** in Zone CP memory).
+If users provide a policy with a correct `kuma.io/zone` tag then no additional conversion is required, but we have to
+validate that zone tag is matching the zone's name. The actual security check is happening on Global but validation is
+helpful to prevent unexpected behaviour.
+
+If users apply a policy on a zone without a `kuma.io/managed-by: zone` label then Zone CP returns an error saying
+"if you want to apply a policy on the zone please add `kuma.io/managed-by: zone` label".
+This measure is going to prevent users form unintentionally applying policies at the wrong place.
```

Comment:
Is the label explicitly added by the user or it's automatically added by a webhook?

Reply:
The idea here is to prevent users from applying things in the wrong place. They probably got used to the fact Zone CP returns an error if you unintentionally run `kubectl apply` against a zone. By requiring the `kuma.io/managed-by: zone` label we're making sure their desire is intentional. So we're **not** going to automate it with a webhook.